Packs/SymantecEDR/Integrations/SymantecEDR/SymantecEDR.py

```diff
@@ -0,0 +1,2734 @@
+        if save_timestamp := global_context.get('access_token_timestamp'):
+            now_timestamp = int(time.time())
+            if token := global_context.get('access_token'):
+                time_diff = int(now_timestamp - save_timestamp)
+                if time_diff <= SESSION_TIMEOUT_SEC:
+                    return token
+                else:
+                    LOG('Access token expired')
+        return None
+
+    def get_access_token_or_login(self) -> None:
+        """
+        Generate Access token
+        """
+        global_context = demisto.getIntegrationContext()
+        if last_access_token := self.get_access_token_from_context(global_context):
+            self.access_token = last_access_token
+            LOG("Access token still active. Re-use the same token")
+        else:
+            try:
+                response = self._http_request(
+                    method='POST',
+                    url_suffix='/atpapi/oauth2/tokens',
+                    auth=(self.client_key, self.secret_key),
+                    data={'grant_type': 'client_credentials'},
+                    resp_type='response'
+                )
+                response.raise_for_status()
+            except requests.exceptions.HTTPError as exc:
+                if error_msg := HTTP_ERRORS.get(exc.response.status_code):
+                    raise DemistoException(f'{error_msg}', res=exc.response) from exc
+                else:
+                    raise
+
+            LOG("Generated Access token.")
+            self.access_token = response.json().get("access_token")
+            timestamp_string = int(time.time())
+            demisto.setIntegrationContext(
+                demisto.getIntegrationContext() | {
+                    'access_token': self.access_token, 'access_token_timestamp': timestamp_string}
+            )
+
+        return None
+
+    def http_request(self, method: str, endpoint: str, params: dict[str, Any] = None,
+                     json_data: Union[dict[str, Any], list] = None,
+                     ignore_empty_response: bool = False) -> requests.Response:
+        """
+        Call Symantec EDR On-prem POST and GET Request API
+        Args:
+            method (str): Request Method support POST and GET
+            endpoint (str): API endpoint
+            params (dict): URL parameters to specify the query for GET.
+            json_data (dict): The dictionary to send in a request for POST.
+            ignore_empty_response (bool): Default is False, For PATCH method provide this argument as True
+        Returns:
+            Return the raw api response from Symantec EDR on-premise API.
+        """
+        try:
+            response = self._http_request(
+                method=method.upper(),
+                url_suffix=endpoint,
+                headers=self.headers,
+                json_data=json_data,
+                params=params,
+                resp_type='response',
+                allow_redirects=False,
```

Comment:
why?

Reply:
I think you are referring to line no 226 "allow_redirects=False," which I'm using explicitly not to redirect to any other location instead get it from the "https:<host>:port" url rather than following chain of redirection which is not handle in this case. If you feel that this is unwanted argument, I can remove it.

Reply:
why is the chain of redirections not handled? is this by design?

Reply:
As discussed, I have removed this argument from _http_request.